cloud/deploy/deploy.go

```diff
@@ -122,7 +123,7 @@ func deployDags(path, runtimeID string, client astro.Client) error {
 }
 
 // Deploy pushes a new docker image
-func Deploy(path, runtimeID, wsID, pytest, envFile, imageName, deploymentName string, prompt, dags bool, client astro.Client) error {
+func Deploy(path, runtimeID, wsID, pytest, envFile, imageName, deploymentName string, prompt, dags bool, client astro.Client) error { //nolint: gocognit
```

Comment:
should we try to simplify the logic in this function to decrease the complexity of the function rather than avoiding linters?

Reply:
I agree. deploy function has too many input args. Got any suggestions to optimize this?

Reply:
a couple of suggestions:
- Have an input struct to take in args for deploy function instead of taking it individually, so that maybe everything except astro client can be part of that struct
- computing `dagDeployEnabled` shouldn't be part of the `buildImage` logic, but could be simply part of Deploy function logic itself and can be computed much early I feel.
- Image deploy logic should be a separate function in itself, just like `deployDags` is at the moment


in the end high level `Deploy` function might look like:
```Golang
func Deploy(input DeployInput, client astro.Client) error {
	// Get cloud domain
	c, err := config.GetCurrentContext()
	if err != nil {
		return err
	}

	domain := c.Domain
	if domain == "" {
		return errors.New("no domain set, re-authenticate")
	}

	if strings.Contains(domain, "cloud") {
		splitDomain := strings.SplitN(domain, ".", splitNum) // This splits out 'cloud' from the domain string
		domain = splitDomain[1]
	}

	dagFiles := fileutil.GetFilesWithSpecificExtension(path+"/dags", ".py")
	if len(dagFiles) == 0 && config.CFG.ShowWarnings.GetBool() {
		i, _ := input.Confirm("Warning: No DAGs found. This will delete any existing DAGs. Are you sure you want to deploy?")

		if !i {
			fmt.Println("Canceling deploy...")
			return nil
		}
	}

         dagDeployEnabled := false
         if strings.HasPrefix(input.runtimeID, "vr-") || input.dags || input.ImageName == ""{
                 dagDeployEnabled = true
         }

        deployInfo, err := getDeploymentInfo(runtimeID, wsID, deploymentName, prompt, domain, client)
	if err != nil {
		return err
	}
	deploymentURL := "cloud." + domain + "/" + deployInfo.workspaceID + "/deployments/" + deployInfo.deploymentID + "/analytics"
        deploymentID := deployInfo.DeploymentID

        if dagDeployEnabled {
		if dags && pytest == allTests { // Not sure if we want to enable pytest for other cases of dag deploy as well or not
			version, _, err := buildImage(&c, path, deployInfo.currentVersion, deployInfo.deployImage, imageName, client)
			if err != nil {
				return err
			}

			err = parseDAG(pytest, version, envFile, deployInfo.deployImage, deployInfo.namespace)
			if err != nil {
				return err
			}
		}

                fmt.Println("Initiating DAGs Deployment for: " + deploymentID)
		err = deployDags(path, deploymentID, client)
		if err != nil {
			return err
		}

		fmt.Println("\nSuccessfully uploaded DAGs to Astro. Navigate to the Airflow UI to confirm that your deploy was successful. The Airflow UI takes about 1 minute to update." +
			"\n\nDeployment can be accessed at the following URLs: \n" +
			fmt.Sprintf("\nDeployment Dashboard: %s", ansi.Bold(deploymentURL)) +
			fmt.Sprintf("\nAirflow Dashboard: %s", ansi.Bold(deployInfo.webserverURL)))
		return nil
       }

       // the entire else logic should be part of `deployImage` I think except `deployDags` part
       err = deployImage(deployImageInput)
	if err != nil {
		return err
	}
	fmt.Println("Successfully pushed Docker image to Astronomer registry. Navigate to the Astronomer UI for confirmation that your deploy was successful." +
		"\n\n Deployment can be accessed at the following URLs: \n" +
		fmt.Sprintf("\n Deployment Dashboard: %s", ansi.Bold(deploymentURL)) +
		fmt.Sprintf("\n Airflow Dashboard: %s", ansi.Bold(deployInfo.webserverURL)))
       return nil
}

Reply:
Agree with points 1 and 3. For point 2, will be refactoring the code a bit. The logic in buildmage needs a different variable name to what there is in imageDeploy, but that logic still needs to be in there for us to be able to switch modes when customers use custom image deploy to dags deploy. The backend mode changes based on what option a user users. Let me clean this up a bit, and you can take another review at this

Reply:
Will put out a separate PR to refactor this.